bokeh/models/widgets/inputs.py

```diff
@@ -51,6 +51,21 @@ class TextInput(InputWidget):
     """)
 
 
+class PasswordInput(InputWidget):
```

Comment:
This needs to extend `TextInput`. Then `callback` and `placeholder` can be removed, as they are already defined in `TextInput`.